prody/proteins/interactions.py

```diff
@@ -0,0 +1,1209 @@
+                listOfHydrogens2 = cleanNumbers(findNeighbors(atoms.hydrogen, 1.4, atoms.select('index '+str(i[1]))))
+                AtomsForAngle = ['D','H','A', 'distance','angle']
+                
+                if not listOfHydrogens1:
+                    for j in listOfHydrogens2:
+                        AtomsForAngle = [j[1], j[0], i[0], i[-1], calcAngle(atoms.select('index '+str(j[1])), 
+                                                                        atoms.select('index '+str(j[0])), 
+                                                                        atoms.select('index '+str(i[0])))[0]]                                                                                   
+                        pairList.append(AtomsForAngle)            
+                
+                elif not listOfHydrogens2:
+                    for jj in listOfHydrogens1:
+                        AtomsForAngle = [jj[1], jj[0], i[1], i[-1], calcAngle(atoms.select('index '+str(jj[1])), 
+                                                                          atoms.select('index '+str(jj[0])), 
+                                                                          atoms.select('index '+str(i[1])))[0]]
+                        pairList.append(AtomsForAngle)            
+       
+                else:            
+                    for j in listOfHydrogens2:
+                        AtomsForAngle = [j[1], j[0], i[0], i[-1], calcAngle(atoms.select('index '+str(j[1])), 
+                                                                            atoms.select('index '+str(j[0])), 
+                                                                            atoms.select('index '+str(i[0])))[0]]                                                                                   
+                        pairList.append(AtomsForAngle)
+    
+                    
+                    for jj in listOfHydrogens1:
+                        AtomsForAngle = [jj[1], jj[0], i[1], i[-1], calcAngle(atoms.select('index '+str(jj[1])), 
+                                                                              atoms.select('index '+str(jj[0])), 
+                                                                              atoms.select('index '+str(i[1])))[0]]
+                        pairList.append(AtomsForAngle)
+    
+    HBs_list = []
+    for k in pairList:
+        if 180-angle < float(k[-1]) < 180 and float(k[-2]) < distA:
+            aa_donor = atoms.getResnames()[k[0]]+str(atoms.getResnums()[k[0]])
+            aa_donor_atom = atoms.getNames()[k[0]]+'_'+str(k[0])
+            aa_donor_chain = atoms.getChids()[k[0]]
+            aa_acceptor = atoms.getResnames()[k[2]]+str(atoms.getResnums()[k[2]])
+            aa_acceptor_atom = atoms.getNames()[k[2]]+'_'+str(k[2])
+            aa_acceptor_chain = atoms.getChids()[k[2]]
+            
+            HBs_list.append([aa_donor, aa_donor_atom, aa_donor_chain, aa_acceptor, aa_acceptor_atom, 
```

Comment:
Perhaps you could order it this way round like in the second part of the SHEET header records of the PDB (taken from line 1578 of 2tbv):

`O  LEU A 263   N  VAL A 103`

See http://www.wwpdb.org/documentation/file-format-content/format33/sect5.html#SHEET


This would include splitting resnum and resname, which is probably a good idea anyway.